src/components/application_manager/rpc_plugins/vehicle_info_plugin/test/commands/mobile/subscribe_vehicle_request_test.cc

```diff
@@ -0,0 +1,325 @@
+    mobile_apis::VehicleDataType::VEHICLEDATA_WINDOWSTATUS;
+const std::string kVehicleTypeStr = am::strings::window_status;
+}  // namespace
+
+class SubscribeVehicleRequestTest
+    : public VICommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  SubscribeVehicleRequestTest()
+      : mock_app_(CreateMockApp())
+      , vi_app_extension_ptr_(
+            std::make_shared<vehicle_info_plugin::VehicleInfoAppExtension>(
+                vi_plugin_, *mock_app_))
+      , app_set_lock_ptr_(std::make_shared<sync_primitives::Lock>())
+      , mock_last_state_(std::make_shared<resumption::LastStateWrapperImpl>(
+            std::make_shared<resumption_test::MockLastState>())) {}
+
+ protected:
+  void SubscribeSuccessfully();
+  void SetUp() OVERRIDE {
+    ON_CALL(app_mngr_, GetRPCHandler())
+        .WillByDefault(ReturnRef(mock_rpc_handler_));
+
+    vi_plugin_.Init(app_mngr_,
+                    mock_rpc_service_,
+                    mock_hmi_capabilities_,
+                    mock_policy_handler_,
+                    mock_last_state_);
+    ON_CALL(*mock_app_, AddExtension(vi_app_extension_ptr_));
+    vi_plugin_.OnApplicationEvent(application_manager::plugin_manager::
+                                      ApplicationEvent::kApplicationRegistered,
+                                  mock_app_);
+    ON_CALL(*mock_app_,
+            QueryInterface(vehicle_info_plugin::VehicleInfoAppExtension::
+                               VehicleInfoAppExtensionUID))
+        .WillByDefault(Return(vi_app_extension_ptr_));
+    ON_CALL(*mock_app_, app_id()).WillByDefault(Return(kConnectionKey));
+  }
+
+  MockAppPtr mock_app_;
+  application_manager::AppExtensionPtr vi_app_extension_ptr_;
+  std::shared_ptr<sync_primitives::Lock> app_set_lock_ptr_;
+  vehicle_info_plugin::VehicleInfoPlugin vi_plugin_;
+  application_manager_test::MockRPCHandler mock_rpc_handler_;
+  resumption::LastStateWrapperPtr mock_last_state_;
+};
+
+TEST_F(SubscribeVehicleRequestTest, Run_AppNotRegistered_UNSUCCESS) {
+  CommandPtr command(CreateCommandVI<SubscribeVehicleDataRequest>());
+  EXPECT_CALL(app_mngr_, application(_))
+      .WillOnce(Return(ApplicationSharedPtr()));
+
+  EXPECT_CALL(
+      mock_rpc_service_,
+      ManageMobileCommand(
+          MobileResultCodeIs(mobile_result::APPLICATION_NOT_REGISTERED), _));
+
+  ASSERT_TRUE(command->Init());
+  command->Run();
+}
+
+TEST_F(SubscribeVehicleRequestTest, Run_DataToSubscribeIsNotExisted_UNSUCCESS) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+  (*command_msg)[am::strings::msg_params][am::strings::button_name] =
+      kVehicleTypeStr;
+
+  am::VehicleData data;
+  EXPECT_CALL(mock_message_helper_, vehicle_data())
+      .WillRepeatedly(ReturnRef(data));
+
+  smart_objects::SmartObject empty_hmi_custom_params;
+  ON_CALL(mock_custom_vehicle_data_manager_, CreateHMIMessageParams(_))
```

Comment:
Should we check the parameter?

Reply:
@LuxoftAKutsan check added https://github.com/smartdevicelink/sdl_core/pull/3391/commits/2856a9cfbfd40b464cb9bf80a2fdec13dd716f96